subsys/bluetooth/host/l2cap.c

```diff
@@ -1594,7 +1607,7 @@ static void l2cap_chan_update_credits(struct bt_l2cap_le_chan *chan,
 	/* Restore enough credits to complete the sdu */
 	credits = ((chan->_sdu_len - net_buf_frags_len(buf)) +
 		   (chan->rx.mps - 1)) / chan->rx.mps;
-	credits -= k_sem_count_get(&chan->rx.credits);
+	credits -= atomic_get(&chan->rx.credits);
```

Comment:
Unrelated to this PR, but credits is s16_t above? Shouldn't it be u16_t @Vudentz ?

Reply:
I will check about it, I recall it being change to s16_t for some reason but maybe Im mistaken.

Reply:
I hope it is u16, since I witness some Androids granting 0xffff credits - and they don't mean -1 credits :)

Reply:
This code is only used to restore RX credits, so it is not influenced by the remote, it restore the amount based on the SDU and then subtract to the existing credits to know if more credits would be needed to complete the SDU, the result can be negative which is why we are using s16_t, we could use a s32_t though that could overflow so it needs to do a bound check as well.

Reply:
In the end I just did the bound check separately and used u16_t as it should be.